### PR TITLE
[SPARK-15085][Streaming][Kafka] Rename streaming-kafka artifact

### PR DESCRIPTION
**Pull request by koeninger**
## What changes were proposed in this pull request?

Renaming the streaming-kafka artifact to include kafka version, in anticipation of needing a different artifact for later kafka versions
## How was this patch tested?

Unit tests

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
(@koeninger it seems the last part of the title is truncated)

**Comment by koeninger** [comment]
@rxin ping to make sure this doesn't get lost in the shuffle before 2.0

**Comment by rxin** [comment]
LGTM if tests pass.

cc @JoshRosen for a quick look.

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -340,7 +340,7 @@ object SparkBuild extends PomBuild {
   val mimaProjects = allProjects.filterNot { x =>
     Seq(
       spark, hive, hiveThriftServer, hiveCompatibility, catalyst, repl, networkCommon, networkShuffle, networkYarn,
-      unsafe, testTags, sketch, mllibLocal
+      unsafe, testTags, sketch, mllibLocal, streamingKafka
```
Comment (JoshRosen) [inline project/SparkBuild.scala]: You don't necessarily have to disable MiMa here. Instead, you can override `MimaKeys.previousArtifact` in the `streamingKafka` project.

**Comment by koeninger** [comment]
@JoshRosen let me know if that fix to pass mima is ok with you.  I can add settings for the kafka subproject instead, but this seemed like the most likely place to get noticed and cleaned up once the version changes and it's no longer needed

**Comment by JoshRosen** [comment]
MiMa change seems fine to me.

**Comment by rxin** [comment]
Thanks - merging in master/2.0.

**Pull request closed without merging**